lib/compat/wordpress-6.5/class-wp-script-modules.php

```diff
@@ -226,6 +228,32 @@ public function print_import_map() {
 			}
 		}
 
+		/**
+		 * Prints the necessary script to load import map polyfill for browsers that
+		 * do not support import maps.
+		 *
+		 * @since 6.5.0
+		 */
+		public function print_import_map_polyfill() {
+			$import_map = $this->get_import_map();
+			if ( ! empty( $import_map['imports'] ) ) {
+				$test = 'HTMLScriptElement.supports && HTMLScriptElement.supports("importmap")';
+				$src  = defined( 'IS_GUTENBERG_PLUGIN' ) && IS_GUTENBERG_PLUGIN ? gutenberg_url( '/build/modules/importmap-polyfill.min.js' ) : includes_url( 'js/dist/vendor/wp-polyfill-importmap.min.js' );
+
+				echo (
+				// Test presence of feature...
+				'<script>( ' . $test . ' ) || ' .
```

Comment:
I meant adding the id to this script. Or better, to both. Also, they should have descriptive names. The import map script id is `wp-importmap`, so calling `wp-js-module-importmap` to the polyfill is confusing. It should at least have the `polyfill` word somewhere. 

This is my suggestion, but take also a look at the ids of other polyfills in WP to keep consistency with them:

- `wp-load-importmap-polyfill` for the script that conditionally loads the polyfill.
- `wp-importmap-polyfill` for the polyfill script.

Reply:
But try first what @gziolo suggested in the WP Core PR, maybe this doesn't have to be done manually:

- https://github.com/WordPress/wordpress-develop/pull/5947#pullrequestreview-1845624529

Reply:
The function `wp_get_script_polyfill` does not provide IDs. So in this case we could only provide an ID to the wrapper, not the script.

Should we make an update to Core to that function to be able to allow IDs?